src/Instance/InstanceMovement/InstanceMovementContainer.js

```diff
@@ -39,15 +39,15 @@ const InstanceMovementContainer = ({
   } = useInstance(idTo, mutator.movableInstance);
 
   const onClose = useCallback((closedInstance) => {
-    const instanceId = closedInstance.id === instanceFrom.id
-      ? instanceTo.id
-      : instanceFrom.id;
+    const instanceId = closedInstance.id === instanceFrom?.id
+      ? instanceTo?.id
+      : instanceFrom?.id;
 
     history.push({
       pathname: `/inventory/view/${instanceId}`,
       search: location.search,
     });
-  });
+  }, [history, location, instanceFrom, instanceTo]);
```

Comment:
@axelhunn the `history` and `location` and `useCallback` doesn't use deep comparison

https://github.com/facebook/react/issues/14476

so this won't work as expected (it will basically recreate `onClose` more often). I would suggest either removing them from the dependency array or using something similar to:

https://github.com/kotarella1110/use-custom-compare

Reply:
Thanks for pointing this out. In fact I made this change for linter not to yell at me.
It happens the `history` is a mutable object (see https://reactrouter.com/web/api/history).
It is the same object every time, so we can rely on === comparison.

As for the `location` - it's not needed here, we can use `history.location` instead,

I've refactored the thing.